src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -434,6 +434,18 @@ internal bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter conso
         private bool ReportErrors(DiagnosticBag diagnostics, TextWriter consoleOutput, ErrorLogger errorLoggerOpt)
             => ReportErrors(diagnostics.ToReadOnly(), consoleOutput, errorLoggerOpt);
 
+        internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
+        {
+            foreach (var diag in diagnostics.AsEnumerableWithoutResolution())
```

Comment:
>AsEnumerableWithoutResolution [](start = 45, length = 29)

Why without resolution? What if we there is a lazy diagnostics that becomes an error after resolution?

Reply:
My understanding is that the next line causes resolution (checks severity) anyways.

Reply:
> My understanding is that the next line causes resolution (checks severity) anyways.

Then why pretend that we do not resolve?

---
In reply to: [266165592](https://github.com/dotnet/roslyn/pull/34174#discussion_r266165592) [](ancestors = 266165592)

Reply:
I'll let Andy respond, but my assumption in reading the PR was to avoid extra processing which `AsEnumerable` does. 
But I agree it's not obvious ;-)

Reply:
Yes, it was for perf and the places I was looking at it I thought it was safe because we always realized diagnostics later, but it's not particularly worth the risk. I've changed it to resolve.